Tools/AP_Periph/AP_Periph.cpp

```diff
@@ -168,7 +168,9 @@ void AP_Periph_FW::init()
 #endif
 
 #ifdef HAL_PERIPH_ENABLE_MSP
-    msp_init(hal.serial(2));
+    if (g.msp_port >= 0) {
+        msp_init(hal.serial(g.msp_port));
```

Comment:
This needs to check for hal_serial_num, or even better to not send a nullptr uart

Reply:
fixed, thanks